docs/ai/conceptual/understanding-tokens.md

```diff
@@ -0,0 +1,118 @@
+
+Once trained, a model can calculate an embedding for a text containing multiple tokens. The model tokenizes the text, then calculates an overall embeddings value based on the learned embeddings of the individual tokens. This technique can be used for semantic document searches or adding [memories](/semantic-kernel/memories/) to an AI.
+
+During output generation, the model predicts a vector value for the next token in the sequence. The model then selects the next token from it's vocabulary based on this vector value. In practice, the models calculates multiple vectors using various elements of the preceding tokens' embeddings. The model then evaluates all potential tokens from these vectors and selects the most probable one to continue the sequence.
+
+Output generation is an iterative operation. The model appends the predicted token to the sequence so far and uses that as the input for the next iteration, building the final output one token at a time.
+
+### Token limits
+
+LLMs have limits on the maximum number of tokens that can be used as input or generated as output. This limit often combines the input and output tokens into a maximum context window.
+
+For example, GPT-4 supports up to 8,192 tokens of context. The combined size of the input and output tokens can't exceed 8,192.
+
+Together a model's token limit and tokenization method determine the maximum length of text that can be provided as input or generated as output.
+
+For example, consider a model with a maximum context window of 100 tokens. For the following input text:
+
+> I heard a dog bark loudly at a cat
+
+With a word-based tokenization method, the input is 9 tokens leaving 91 **word** tokens available for the output.
+
+With a character-based tokenization method, the input is 34 tokens (including spaces) leaving only 66 **character** tokens available for the output.
+
+### Token based pricing and rate limiting
+
+Generative AI services often use token-based pricing. The cost of each request depends on the number of input and output tokens. The pricing might differ between input and output. For example, see [Azure OpenAI Service pricing](https://azure.microsoft.com/pricing/details/cognitive-services/openai-service/).
+
+Generative AI services might also have limits on the maximum number of tokens per minute (TPM). These rate limits can vary depending on the service region and LLM. For more information about specific regions, see [Azure OpenAI Service quotas and limits](/azure/ai-services/openai/quotas-limits#regional-quota-limits).
+
+## Related content
+
+- [How Generative AI and LLMs work]<!-- (how-genai-and-llms-work.md) -->
+- [Understanding embeddings]<!-- (understanding-embeddings.md) -->
+- [Working with vector databases]<!-- (working-with-vector-dbs.md) -->
```

Comment:
```suggestion
- [Working with vector databases]<!-- (working-with-vector-dbs.md) -->

[!INCLUDE [Third-party disclaimer](../../includes/third-party-contact-disclaimer.md)]
```
This is required because of the link to the OpenAI website (for the tokenizer tool). This disclaimer (and the third-party information disclaimer) goes at the end of the last public section.

Reply:
There isn't a doc by that name in this repo. @gewarren is there a standard include for third-party disclamers?